joern-cli/frontends/solidity2cpg/src/main/scala/io/joern/solidity2cpg/passes/AstCreator.scala

```diff
@@ -648,80 +589,93 @@ class AstCreator(filename: String, sourceUnit: SourceUnit, global: Global) exten
   }
 
   private def astForField(stateVariableDeclaration: StateVariableDeclaration, order: Int): Ast = {
-    val fieldType =
+    val memberAsts =
       stateVariableDeclaration.variables.collect { case x: VariableDeclaration => astForMemberVarDecl(x, order) }
 
-    Ast().withChildren(fieldType)
+    // TODO Return a sequence of Asts here.
+    Ast().withChildren(memberAsts)
   }
 
   private def getMappingKeyAndValue(mapping: Mapping): String = {
+    // TODO Are these matches comprehensive?
```

Comment:
We could consider constructing a new trait from what we expect to come from `Mapping` examples and have a better bound of expected types.

Reply:
This is again something to add to the case class construction side